crypto/s2n_fips.c

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright 2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include <openssl/crypto.h>
+
+#include "crypto/s2n_fips.h"
+
+/* Return 1 if FIPS mode is enabled, 0 otherwise. */
+int s2n_is_in_fips_mode()
+{
+#ifdef OPENSSL_FIPS
+    /* FIPS mode can be entered only if OPENSSL_FIPS is defined */
+    return !!FIPS_mode();
```

Comment:
The "!!" doesn't seem very readable to me and looks like it could be easily missed if you're not paying attention. How do others feel? 

In what cases does `FIPS_mode()` return something outside of `0` and `1`?

I'd rather go the more explicit (and verbose) route and do:
```
if (FIPS_mode()) {
  return 1;
} else {
  return 0;
}
```
to guarantee that only `0` or `1` is returned and not some other value.